datafusion/expr-common/src/groups_accumulator.rs

```diff
@@ -106,6 +107,44 @@ impl EmitTo {
 /// [`Accumulator`]: crate::accumulator::Accumulator
 /// [Aggregating Millions of Groups Fast blog]: https://arrow.apache.org/blog/2023/08/05/datafusion_fast_grouping/
 pub trait GroupsAccumulator: Send {
+    /// Whether [`Self::with_group_indices_order_mode`] should be called.
+    ///
+    /// this is when the accumulator would benefit from knowing the order of the group indices.
+    ///
+    fn group_order_sensitivity(&self) -> bool {
```

Comment:
Minor: I think a name like `supports_with_input_order_mode` is more consistent with other methods on this trait such as `supports_convert_to_state`

Reply:
the problem with `*_input_order_mode` gives the sense that the input batches are sorted while they are not

Reply:
I think having the `supports` method have most of the same name as the method that is called makes the most sense

So for example, if other function is called in `with_group_indices_order_mode` I think this method should be named consistently like `supports_with_group_indices_order_mode`

Reply:
updated